### PR TITLE
Fix/create lead page bugs

### PR DESCRIPTION
**Pull request by AKutusenko**
## Describe your changes

- Fixed offering type bug
- Changed placeholders
- Fixed price range bug
- Fixed ploor area bug
- Moved error messages to i18n

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-248)
-  [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-240)
- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-247)
- [Link to JIRA ticket](https://testhomework4.atlassian.net/jira/software/projects/AG/boards/6?assignee=5fd26b3708757f006e4d61d4&selectedIssue=AG-246)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [ ] No inline styles
- [x] Imports are absolute
- [ ] Attach a screenshot if PR has visual changes.

### CODE REVIEW

**Review by canterbery — APPROVED** [review]

**Review by canterbery — APPROVED** [review]

File: src/sections/create-lead-view/form.tsx [hunk 1]
```diff
@@ -75,46 +73,54 @@ export default function Form({ user }: Props): JSX.Element {
...
   const watchAllFields = watch();
 
   useEffect(() => {
-    setIsEnquiryTypeRent(watchAllFields.offeringType?.value === 'to_rent');
-    setIsEnquiryTypeSell(watchAllFields.offeringType?.value === 'to_sell');
-  }, [watchAllFields.offeringType?.value]);
+    setValue('priceRangeRentFrom', 100);
+    setValue('priceRangeRentTo', 10000);
+    setValue('priceRangeBuyFrom', 10000);
+    setValue('priceRangeBuyTo', 10000000);
```
Comment (lyudad) [inline src/sections/create-lead-view/form.tsx]: magic numbers

File: src/sections/create-lead-view/form.tsx [hunk 2]
```diff
@@ -127,35 +133,39 @@ export default function Form({ user }: Props): JSX.Element {
...
+    } else {
+      requestData.list_rental_price_from = priceRangeRentFrom;
+      requestData.list_rental_price_to = priceRangeRentTo;
+    }
+
     try {
       await createLead(requestData).unwrap();
       push(AppRoute.LEADS_PAGE);
-      reset();
+      reset(defaultValues);
```
Comment (lyudad) [inline src/sections/create-lead-view/form.tsx:170]: move all logic to custom hook

File: src/sections/create-lead-view/schema.ts [hunk 3]
```diff
@@ -21,17 +23,148 @@ export type Shape<Fields> = {
...
 };
 
+const RANGES = {
+  totalAreaMin: 0,
+  totalAreaMax: 10000,
+  priceRangeRentMin: 100,
+  priceRangeRentMax: 10000,
+  priceRangeBuyMin: 10000,
+  priceRangeBuyMax: 10000000,
+};
```
Comment (lyudad) [inline src/sections/create-lead-view/schema.ts]: if you have this numbers if why you don't reuse it?

File: src/sections/create-lead-view/form.tsx (hunk 1 above)
Comment (AKutusenko) [reply src/sections/create-lead-view/form.tsx]: fixed

File: src/sections/create-lead-view/schema.ts (hunk 3 above)
Comment (AKutusenko) [reply src/sections/create-lead-view/schema.ts]: fixed

File: src/sections/create-lead-view/form.tsx (hunk 2 above)
Comment (AKutusenko) [reply src/sections/create-lead-view/form.tsx:170]: As i know we cant use hooks in callback functions?

**Review by anette1983 — APPROVED** [review]

**Pull request merged**